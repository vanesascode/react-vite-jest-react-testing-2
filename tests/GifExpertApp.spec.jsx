import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Tests in <GifExpertApp />", () => {
  test("should match with snapshot", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test("Should render the title", () => {
    render(<GifExpertApp />);

    screen.debug();

    const title = screen.getByRole("heading", {
      level: 1,
      name: "GifExpertApp",
    });

    expect(title).toBeTruthy();
  });

  test("should add to the categories array when the form is submitted", () => {
    render(<GifExpertApp />);
    const input = screen.getByRole("textbox");

    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: "test" } });

    fireEvent.submit(form);

    const cardGridElements = screen.getAllByTestId("grid");

    expect(cardGridElements.length).toBe(2);
  });
});
