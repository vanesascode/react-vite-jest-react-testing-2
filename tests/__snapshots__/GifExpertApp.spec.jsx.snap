// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    cats
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
    data-testid="grid"
  />
  u
</div>
`;
